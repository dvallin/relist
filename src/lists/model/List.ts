export interface List {
    name: string
    items: string[]
}
