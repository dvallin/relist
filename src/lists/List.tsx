import * as React from 'react'

import { List } from './model/List'

import { ListGroup, ListGroupItem } from 'react-bootstrap'

export interface ListProps {
    list: List
}

class ListComponent extends React.Component<ListProps, {}> {
    public render(): JSX.Element {
        const { list } = this.props
        const items = list.items.map((value, index) => <ListGroupItem key={index}>{value}</ListGroupItem>)
        return <ListGroup>{items}</ListGroup>
    }
}

export default ListComponent
