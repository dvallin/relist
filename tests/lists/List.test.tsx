import * as React from 'react'
import List from '../../src/lists/List'
import { mount } from '../test-utils'

describe('List', () => {
    describe('with two items', () => {
        const list = { name: 'listName', items: ['item1', 'item2'] }
        const component = <List list={list} />
        it('renders snapshot', () => {
            const element = mount(component)
            expect(element).toMatchSnapshot()
        })

        it('renders item names of a list', () => {
            const element = mount(<List list={list} />)
            const items = element.find('div.list-group-item')
            expect(items.at(0).text()).toEqual('item1')
            expect(items.at(1).text()).toEqual('item2')
        })
    })
})
