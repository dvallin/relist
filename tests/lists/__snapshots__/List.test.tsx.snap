// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List with two items renders snapshot 1`] = `
<ListComponent
  list={
    Object {
      "items": Array [
        "item1",
        "item2",
      ],
      "name": "listName",
    }
  }
>
  <ForwardRef>
    <Uncontrolled(Component)
      innerRef={null}
    >
      <ForwardRef(Bootstrap(ListGroup))
        onSelect={[Function]}
      >
        <ListGroup
          as="div"
          bsPrefix="list-group"
          onSelect={[Function]}
          variant={null}
        >
          <ForwardRef(ContextTransform(AbstractNav))
            as="div"
            className="list-group"
            onSelect={[Function]}
          >
            <AbstractNav
              as="div"
              className="list-group"
              onSelect={[Function]}
            >
              <div
                className="list-group"
                onKeyDown={[Function]}
              >
                <ForwardRef(Bootstrap(ListGroupItem))
                  key="0"
                >
                  <ListGroupItem
                    active={false}
                    bsPrefix="list-group-item"
                    disabled={false}
                    variant={null}
                  >
                    <AbstractNavItem
                      as="div"
                      className="list-group-item"
                      disabled={false}
                      eventKey={null}
                      onClick={[Function]}
                    >
                      <div
                        className="list-group-item"
                        data-rb-event-key={null}
                        disabled={false}
                        onClick={[Function]}
                      >
                        item1
                      </div>
                    </AbstractNavItem>
                  </ListGroupItem>
                </ForwardRef(Bootstrap(ListGroupItem))>
                <ForwardRef(Bootstrap(ListGroupItem))
                  key="1"
                >
                  <ListGroupItem
                    active={false}
                    bsPrefix="list-group-item"
                    disabled={false}
                    variant={null}
                  >
                    <AbstractNavItem
                      as="div"
                      className="list-group-item"
                      disabled={false}
                      eventKey={null}
                      onClick={[Function]}
                    >
                      <div
                        className="list-group-item"
                        data-rb-event-key={null}
                        disabled={false}
                        onClick={[Function]}
                      >
                        item2
                      </div>
                    </AbstractNavItem>
                  </ListGroupItem>
                </ForwardRef(Bootstrap(ListGroupItem))>
              </div>
            </AbstractNav>
          </ForwardRef(ContextTransform(AbstractNav))>
        </ListGroup>
      </ForwardRef(Bootstrap(ListGroup))>
    </Uncontrolled(Component)>
  </ForwardRef>
</ListComponent>
`;
